include/zephyr/drivers/location.h

```diff
@@ -0,0 +1,338 @@
+		return -EINVAL;
+	}
+
+	if (((const struct location_driver_api *)dev->api)->bearing_get == NULL) {
+		return -ENOSYS;
+	}
+
+	return ((const struct location_driver_api *)dev->api)->bearing_get(dev, bearing);
+}
+
+/**
+ * @brief Get the latest speed from the device instance
+ *
+ * @param dev Device instance
+ * @param speed Destination for speed data
+ * @return -EINVAL if any argument is NULL
+ * @return -ENOSYS if api call is not implemented by driver
+ * @return -ENOENT if no data is available
+ * @return 0 if successful, negative errno code otherwise
+ */
+__syscall int location_speed_get(const struct device *dev, struct location_speed *speed);
+
+static inline int z_impl_location_speed_get(const struct device *dev,
+			struct location_speed *speed)
+{
+	if (dev == NULL || speed == NULL) {
+		return -EINVAL;
+	}
+
+	if (((const struct location_driver_api *)dev->api)->speed_get == NULL) {
+		return -ENOSYS;
+	}
+
+	return ((const struct location_driver_api *)dev->api)->speed_get(dev, speed);
+}
+
+
+/**
+ * @brief Get the latest altitude from the device instance
+ *
+ * @param dev Device instance
+ * @param altitude Destination for altitude data
+ * @return -EINVAL if any argument is NULL
+ * @return -ENOSYS if api call is not implemented by driver
+ * @return -ENOENT if no data is available
+ * @return 0 if successful, negative errno code otherwise
+ */
+__syscall int location_altitude_get(const struct device *dev, struct location_altitude *altitude);
+
+static inline int z_impl_location_altitude_get(const struct device *dev,
+			struct location_altitude *altitude)
+{
+	if (dev == NULL || altitude == NULL) {
+		return -EINVAL;
+	}
+
+	if (((const struct location_driver_api *)dev->api)->altitude_get == NULL) {
+		return -ENOSYS;
+	}
+
+	return ((const struct location_driver_api *)dev->api)->altitude_get(dev, altitude);
+}
+
+/**
+ * @brief Set the location event handler
+ *
+ * @param dev Device instance
+ * @param handler The handler to invoke when a location event occurs, can be set to NULL
+ * @param user_data User data provided to handler when it is invoked
+ * @return -EINVAL if device is NULL
+ * @return -ENOSYS if api call is not implemented by driver
+ * @return 0 if successful, negative errno code otherwise
+ */
+__syscall int location_event_handler_set(const struct device *dev,
+			location_event_handler_t handler, void *user_data);
+
+static inline int z_impl_location_event_handler_set(const struct device *dev,
+			location_event_handler_t handler, void *user_data)
+{
+	if (dev == NULL) {
+		return -EINVAL;
+	}
+
+	if (((const struct location_driver_api *)dev->api)->event_handler_set == NULL) {
+		return -ENOSYS;
+	}
+
+	const struct location_driver_api *api = (const struct location_driver_api *)dev->api;
+
+	return api->event_handler_set(dev, handler, user_data);
+}
+
+/**
+ * @}
+ */
+
+/**
+ * @brief Location API macros
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#define LOCATION_DEVICE_API_DEFINE(node_id, api_ptr) \
```

Comment:
Not seeing any code that uses any of these macros.

Reply:
This is just a draft :) The real PR will be created once we agree on the design ;)